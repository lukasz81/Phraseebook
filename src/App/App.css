.flex-container {
  display: flex;
  align-items: center;
  justify-content: flex-start; }

.flex-row {
  flex-direction: row; }

.default-padding {
  padding: 10px; }

svg {
  vertical-align: top; }

body {
  margin: 0;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #777; }
  body p {
    margin: 0; }
  body .profile-picture {
    display: block;
    width: 50px;
    height: 50px;
    margin-right: 10px;
    border-radius: 2px; }
  body ul {
    list-style-type: none;
    margin: 0;
    padding: 0; }
    body ul li {
      box-shadow: 0 1px 0 0 #ddd;
      padding: 10px 0;
      transition: all 0.2s ease-in-out; }
      body ul li.notification-container {
        cursor: pointer; }
        body ul li.notification-container:hover {
          color: #333; }
        body ul li.notification-container .two-line-text {
          overflow: hidden;
          position: relative;
          line-height: 1.2em;
          max-height: 2.4em;
          text-align: left;
          margin-right: -1em;
          padding-right: 1em; }
          body ul li.notification-container .two-line-text:before {
            content: '...';
            position: absolute;
            right: 0;
            bottom: 0; }
          body ul li.notification-container .two-line-text:after {
            content: '';
            position: absolute;
            right: 0;
            width: 1em;
            height: 1em;
            margin-top: 0.2em;
            background: white; }
