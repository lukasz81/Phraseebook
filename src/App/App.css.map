{
"version": 3,
"mappings": "AAAA,IAAK;EACH,MAAM,EAAE,CAAC;EACT,OAAO,EAAE,CAAC;EACV,WAAW,EAAE,iJAAiJ;EAC9J,sBAAsB,EAAE,WAAW;EACnC,uBAAuB,EAAE,SAAS",
"sources": ["App.scss"],
"names": [],
"file": "App.css"
}