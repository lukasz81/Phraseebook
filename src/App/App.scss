@import "../colors.scss";
@import "../mixins.scss";

.flex-container {
  display: flex;
  align-items: center;
  justify-content: flex-start;
}
//.flex-row {
//  flex-direction: row;
//}
.space-between {
  justify-content: space-between;
}
.default-padding {
  padding: 10px;
}

svg {
  vertical-align: top;
}

body {
  margin: 0;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: $body-text-color;
  p{
    margin:0;
    line-height: 1.5;
  }
  .profile-picture {
    display: block;
    width: 50px;
    height: 50px;
    margin-right: 10px;
    border-radius: 2px;
  }
  ul {
    list-style-type: none;
    margin:0;
    padding:0;
    li {
      box-shadow: 0 1px 0 0 #ddd;
      padding: 10px 0;
      transition: all 0.2s ease-in-out;
      &.notification-container {
        cursor: pointer;
        &:hover {
          color: $main-color-hover
        }
        .two-line-text {
          @include multiLineEllipsis($lineHeight: 1.2em, $lineCount: 2, $bgColor: white);
        }
      }
    }
  }
}