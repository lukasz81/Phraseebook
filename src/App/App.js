import React, {Component} from 'react';
import NotificationsList from '../NotificationsList/NotificationsList';
import NavBar from '../NavBar/NavBar';
import './App.css';
import {Route, Switch} from "react-router-dom";
import {connect} from 'react-redux';
import { MdHome, MdNotifications } from 'react-icons/md';
import {fetchNotificationsAction} from '../NotificationsList/notificationsActions';
const lorem = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.'

export class App extends Component {

    componentWillMount() {
        // Load content if not in store.
        if (!this.props.notifications.length) this.props.fetchNotificationsAction()
    }

    contentForDetailsPage(props) {
        return (
            <div className={'default-padding details-page'}>
                <h3><MdNotifications size={20}/> Notification</h3>
                <p>Notification content for post: {props.match.params.id}</p>
            </div>
        )
    };

    contentForHomePage() {
        return (
            <div className={'default-padding home-page'}>
                <h3><MdHome size={20}/> Home</h3>
                <p>{lorem}</p>
            </div>
        )
    };

    render() {
        const {history} = this.props;
        return (
            <div className="App">
                <NavBar history={history} props={this.props}/>
                <main className={'full-height bg-white default-width'}>
                    <Switch>
                        <Route history={history} exact path="/" render={() => this.contentForHomePage()}/>
                        <Route history={history} path="/notifications" component={NotificationsList}/>
                        <Route history={history} path='/notification/:id' render={ props => this.contentForDetailsPage(props)}/>
                    </Switch>
                </main>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        isWaiting: state.isWaiting,
        notifications: state.notifications
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchNotificationsAction: () => {
            fetchNotificationsAction(dispatch)
        }
    }
};

export default connect(mapStateToProps,mapDispatchToProps)(App);