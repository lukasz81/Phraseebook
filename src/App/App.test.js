import {App} from './App';
import Enzyme, {shallow} from "enzyme";
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

describe('Tests button component',() => {

    const makeComponent = (loggedIn = false) => {
        return shallow(
            <App
                fetchNotificationsAction={() => {}}
                isLoggedIn={loggedIn}
                notifications={[]}
            />
        )
    };

    it('should render correctly',() => {
        expect(makeComponent()).toMatchSnapshot();
    });

});
