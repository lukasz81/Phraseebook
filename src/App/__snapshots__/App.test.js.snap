// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests button component should render correctly 1`] = `
<div
  className="App"
>
  <withRouter(NavBar)
    props={
      Object {
        "fetchNotificationsAction": [Function],
        "isLoggedIn": false,
        "notifications": Array [],
      }
    }
  />
  <main
    className="full-height bg-white default-width"
  >
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        component={[Function]}
        path="/notifications"
      />
      <Route
        path="/notification/:id"
        render={[Function]}
      />
    </Switch>
  </main>
</div>
`;
